### DISCUSSION TITLE
Alert - Single query with multiple columns in expressions and alert description

### DISCUSSION THREAD

**Discussion by kounoupaki** [grafana/grafana-infinity-datasource#955] [Q&A]
Hello all! I am trying to create an alert using infinity datasource backend parser. I have a single query/rest api call that returns multiple metrics in json. So I use selectors to map the metrics to multiple columns. I use calculated columns to create the column with the alert condition and using summarize I can use the condition column on the expression field and mark it as Alert condition. The alert seems to work OK. However, I am not able to parse the other columns with the metrics in order to include them in the description field of the notification and other annotations. Is it possible in any way? The only workaround I have found is to create separate queries/api calls but this is not ideal since the whole information required is included in one api call.

**Comment by morangux** [comment]
Can you show a screen shot of the alert rule ? How did you configure it ?

**Reply by kounoupaki** [reply]
@morangux A sample response from the API is the following:
```json
{
	"timestamp": 1.723119570012E9,
	"metric_1": 1.0,
	"metric_2": 49.0,
	"metric_3": -7.0
}
```
I configure the alert with computed columns:
![grafana_alert](https://github.com/user-attachments/assets/7e1a68eb-e663-48af-bdc7-e266b524924e)
If I want to create an alert condition combining multiple metrics eg metric1a, metric2a and metric3a the only way I can find is to create 3 x separate queries/api calls and use sum(metric1a) etc in the summarize section of each query. 
Is it possible to use logical operations (and, or) on a computed column expression? 
Is it possible to access values of the columns & computed columns of the query in the alert expressions section where the alert condition is configured without creating separate query for each one of them?
